### DISCUSSION TITLE
templateddialog namespace

### DISCUSSION THREAD

**Discussion by shakazulu89** [serenity-is/Serenity#6804] [Help] [answered]
hey
so running into an issue when i have a templated dialog in a folder where the folder name is different from the dialog namespace.
for example, i have a folder **"Listings"**. inside i have a templated dialog **"SelectCaseTypeDialog"**. the namespace for this dialog is **"ProjectXYZ.Dialogs"** . (i wanted to put all my dialogs in that namespace so its easy to reference any dialog instead of having to remember different namespace and where the dialog is).

now the dialog has a template file **"SelectCaseTypeDialog.Template.html"** .
when i try to open this dialog from anywhere i get the error:

> "Script data Template.SelectCaseTypeDialog is not found in registered script list!"

if i run in console "**Q.ScriptData.ensure('Template.Listings.SelectCaseTypeDialog')**" it finds the dialog template.

somewhere its forcing it to script the template in **"Listings"** folder.  i dont understand why it doesnt script it in **'Template.Dialogs.SelectCaseTypeDialog'.**..

i tried using 
`@Serenity.Decorators.registerClass("Dialogs.SelectCaseTypeDialog")`

hoping that will force it to scriptdata there but not sure what registerclass actually does?..

my question: _is there anyway around this or must i have my dialog in a namespace with the same name as its parent folder?_

thanks for any feedback!

**Comment by VictorTomaili** [comment, marked as answer]
Override the `getTemplateName` on your widget if the namespace and folder names are different.
